truffle/contracts/VaultNotary/VaultNotary.sol

```diff
@@ -0,0 +1,231 @@
+/// @author Jianwei Liu jliu@shipchain.io
+pragma solidity 0.5.11;
+
+import {Ownable} from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+
+/** @notice The VaultNotary contract is the contract for reading/writing the
+  * vault uri and hash, and controlling the permissions to those operations by vault
+  * owner.
+  */
+contract VaultNotary is Ownable {
+    struct Data {
+        // The address of the Vault owner
+        address vaultOwner;
+        string vaultHash;
+        string vaultUri;
+
+        // The access control mapping to record whether an address can update the Uri of a Vault
+        mapping(address => bool) aclUriMapping;
```

Comment:
Since the type is `mapping` I think it would be cleaner to just name this something like uriAcl or uriUpdateAcl
```suggestion
        mapping(address => bool) uriAcl;
```